sentry-core/src/main/java/io/sentry/core/transport/AsyncConnection.java

```diff
@@ -159,17 +167,25 @@ public void run() {
                     + "ms.";
 
             if (options.isDebug()) {
-              options.getLogger().log(SentryLevel.DEBUG, message);
+              options.getLogger().log(SentryLevel.ERROR, message);
             }
 
             throw new IllegalStateException(message);
           }
         } catch (IOException e) {
           eventCache.store(event);
+          // Failure due to IO is allowed to retry the event
+          if (hint instanceof io.sentry.core.hints.Retryable) {
```

Comment:
we already have a `Retryable` object, let's merge it but keep in mind to change this latter.